### ISSUE TITLE
Invisible UI Images when using certain transforms

### ISSUE THREAD

**Issue by Roryl-c** [bevyengine/bevy#5293]
## Bevy version

a1d3f1b3b42c3a6e05a532362ddfccf0be6b5df3

This **does not reproduce** on current stable version 0.7

## Relevant system information

```ignore
AdapterInfo { name: "NVIDIA GeForce RTX 3070", vendor: 4318, device: 9348, device_type: DiscreteGpu, backend: Vulkan }
```

## What you did

Spawned a UI image with a transform (e.g. to flip a UI sprite using rotation or scale):
```rust
    commands.spawn_bundle(Camera2dBundle::default());
    commands.spawn_bundle(ImageBundle {
        style: Style {
            size: Size::new(Val::Px(500.0), Val::Px(500.0)),
            ..default()
        },
        image: asset_server.load("branding/bevy_logo_dark_big.png").into(),
        transform: Transform::from_rotation(Quat::from_rotation_x(1.0)),
        ..default()
    });
```

## What went wrong

Depending on the transform, the image either renders clipped or does not render at all.

## Additional information

`transform: Transform::from_rotation(Quat::from_rotation_z(0.5)),` (rotates with distortion on both)
![image](https://user-images.githubusercontent.com/5150427/178401033-23f7d0cd-e16c-41ad-a591-bad93b1fa1ce.png)

`transform: Transform::from_rotation(Quat::from_rotation_z(1.0)),` (does not render on main)
![image](https://user-images.githubusercontent.com/5150427/178400894-39480863-5040-4a6b-ba53-58c32dd3855a.png)

`transform: Transform::from_scale(vec3(1.0, 0.5, 1.0)),` (scales correctly on both)
![image](https://user-images.githubusercontent.com/5150427/178400745-aec14a91-1ba3-4d50-b47a-8603baa9d6fa.png)

`transform: Transform::from_scale(vec3(1.0, -0.5, 1.0)),` (does not render on main)
![image](https://user-images.githubusercontent.com/5150427/178400576-fd37d10c-de46-4514-8fb5-05d899ce8d6e.png)

**Comment by alice-i-cecile** [comment]
@superdump @cart I think we need to ensure this is fixed by 0.8; this is a severe regression.

**Comment by Roryl-c** [comment]
The clipping with x-rotation previously included in the issue description is actually not a regression, the same behavior is exhibited on 0.7. So the only regression here is the images going invisible with certain transforms.

**Comment by mockersf** [comment]
I tried a quick fix for this in #5389
